ui/ozone/platform/wayland/wayland_buffer_manager.cc

```diff
@@ -105,9 +105,17 @@ bool WaylandBufferManager::CreateBuffer(base::File file,
 
   uint32_t fd = file.TakePlatformFile();
   for (size_t i = 0; i < planes_count; i++) {
-    zwp_linux_buffer_params_v1_add(params, fd, i /* plane id */, offsets[i],
-                                   strides[i], modifiers[i] >> 32,
-                                   modifiers[i] & UINT32_MAX);
+    if (modifiers[i] == DRM_FORMAT_MOD_INVALID) {
+      DCHECK_EQ(planes_count, 1u) << "Invalid modifier may be passed only in "
+                                     "case of single plane format being used";
+      zwp_linux_buffer_params_v1_add(params, fd, i /* plane id */, offsets[i],
+                                     strides[i], 0, 0);
+
+    } else {
+      zwp_linux_buffer_params_v1_add(params, fd, i /* plane id */, offsets[i],
+                                     strides[i], modifiers[i] >> 32,
+                                     modifiers[i] & UINT32_MAX);
+    }
```

Comment:
This way we'll have two branches with similar calls to `zwp_linux_buffer_params_v1_add`.  I'd suggest reverting this to the original version and adding a DCHECK like this:
`DCHECK(modifiers[i] != DRM_FORMAT_MOD_INVALID || planes_count == 1u) << "message about invalid mod"`

Reply:
discussed offline.